wab/src/main/java/org/apache/unomi/web/ContextServlet.java

```diff
@@ -71,396 +53,21 @@ public void init(ServletConfig config) throws ServletException {
-        if (filterNodes != null) {
-            data.setFilteringResults(new HashMap<>());
-            for (PersonalizationService.PersonalizedContent personalizedContent : sanitizePersonalizedContentObjects(filterNodes)) {
-                data.getFilteringResults().put(personalizedContent.getId(), personalizationService.filter(profile,
-                        session, personalizedContent));
-            }
-        }
-
-        List<PersonalizationService.PersonalizationRequest> personalizations = contextRequest.getPersonalizations();
-        if (personalizations != null) {
-            data.setPersonalizations(new HashMap<>());
-            for (PersonalizationService.PersonalizationRequest personalization : sanitizePersonalizations(personalizations)) {
-                data.getPersonalizations().put(personalization.getId(), personalizationService.personalizeList(profile,
-                        session, personalization));
-            }
-        }
-
-        if (!(profile instanceof Persona)) {
-            data.setTrackedConditions(rulesService.getTrackedConditions(contextRequest.getSource()));
-        } else {
-            data.setTrackedConditions(Collections.emptySet());
-        }
-
-        data.setAnonymousBrowsing(privacyService.isRequireAnonymousBrowsing(profile));
-        data.setConsents(profile.getConsents());
-
-        return changes;
-    }
-
-    /**
-     * This function will update the profile if it is from Persona instance.
-     * The profile will be updated using the overrides attributes :
-     * - profileOverrides for profile properties, segments and scores
-     * - sessionPropertiesOverrides for session properties
-     * @param contextRequest
-     * @param profile
-     * @param session
-     */
-    private void processOverrides(ContextRequest contextRequest, Profile profile, Session session) {
-        if (profile instanceof Persona) {
-            if (contextRequest.getProfileOverrides() != null) {
-                if (contextRequest.getProfileOverrides().getScores()!=null) {
-                    profile.setScores(contextRequest.getProfileOverrides().getScores());
-                }
-                if (contextRequest.getProfileOverrides().getSegments()!=null) {
-                    profile.setSegments(contextRequest.getProfileOverrides().getSegments());
-                }
-                if (contextRequest.getProfileOverrides().getProperties()!=null) {
-                    profile.setProperties(contextRequest.getProfileOverrides().getProperties());
-                }
-                if (contextRequest.getSessionPropertiesOverrides()!=null && session != null) {
-                    session.setProperties(contextRequest.getSessionPropertiesOverrides());
-                }
-            }
+            response.flushBuffer();
+            logger.debug("OPTIONS request received. No context will be returned.");
+            return;
```

Comment:
Options should also be implemented in the REST endpoint, so like the event collector, we could just forward the request and implement the option in the endpoint